synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -185,13 +187,18 @@ async def _get_power_levels_and_sender_level(
 
         return pl_event.content if pl_event else {}, sender_level
 
+    @measure_func("action_for_event_by_user")
     async def action_for_event_by_user(
         self, event: EventBase, context: EventContext
     ) -> None:
         """Given an event and context, evaluate the push rules, check if the message
         should increment the unread count, and insert the results into the
         event_push_actions_staging table.
         """
+        if event.internal_metadata.is_outlier():
+            # This can happen due to out of band memberships
+            return
```

Comment:
@erikjohnston are we OK with this moving into the measured function? It was added in #12689.

Reply:
Yeah, that's fine!